lib/multiplexer/multiplexer.go

```diff
@@ -414,75 +433,102 @@ func (m *Mux) detect(conn net.Conn) (*Conn, error) {
 
 		switch proto {
 		case ProtoProxy:
-			if !m.EnableExternalProxyProtocol {
+			newPROXYLine, err := ReadProxyLine(reader)
+			if err != nil {
+				return nil, trace.Wrap(err, invalidProxyLineError)
+			}
+
+			if m.PROXYProtocolMode == PROXYProtocolOff {
 				return nil, trace.BadParameter(proxyProtocolDisabledError)
 			}
-			// We allow only one unsigned proxy line
-			if proxyLine != nil {
-				return nil, trace.BadParameter(duplicateProxyLineError)
+
+			unsignedPROXYLineReceived = true
+
+			if m.PROXYProtocolMode == PROXYProtocolUnspecified {
+				m.Logger.WithFields(log.Fields{
```

Comment:
I think we should rate this log a bit otherwise we're going to spam a lot.

Reply:
I personally am ok with spamming it, that's kinda the point. But I guess we can rate limit them as well. We already have log limiter in the multiplexer, it deduplicates logs within 1 minute, should I change the code to use it?

Reply:
We should rate limit.